### PR TITLE
Xcode version update

### PR DESCRIPTION
**Pull request by stechiu**
### Summary of changes

 - Bumped Xcode version to 15.0

https://developer.apple.com/support/xcode/

 ### Checklist

 - [x] Added a changelog entry

- @stechiu

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Braintree iOS Drop-in SDK - Release Notes
 
+## unreleased
+* Require Xcode 15.1+ (per [App Store requirements](https://developer.apple.com/news/?id=khzvxn8a))
```
Comment (jaxdesmarais) [inline CHANGELOG.md:4]:
```suggestion
* Require Xcode 15.0+ (per [App Store requirements](https://developer.apple.com/news/?id=khzvxn8a))
```

File: README.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ Welcome to Braintree's Drop-in SDK for iOS!
 
 ![Drop-in light theme](Images/client-sdk-ios-series-light.png "Drop-in light theme")
 
-**The Braintree iOS Drop-in SDK permits a deployment target of iOS 12.0 or higher.** It requires Xcode 14.1+ and Swift 5.7.1+.
+**The Braintree iOS Drop-in SDK permits a deployment target of iOS 17.2 or higher.** It requires Xcode 15.1+ and Swift 4+.
```
Comment (jaxdesmarais) [inline README.md:12]:
```suggestion
**The Braintree iOS Drop-in SDK permits a deployment target of iOS 12.0 or higher.** It requires Xcode 15.0+ and Swift 5.9+.
```

File: README.md [hunk 3]
```diff
@@ -45,7 +45,7 @@ Then run `pod install`.
 
 See our [`Podspec`](https://github.com/braintree/braintree-ios-drop-in/blob/main/BraintreeDropIn.podspec) for more information.
 
-*Note:* If you are using version 8.x.x of the Braintree Drop-in iOS SDK in Xcode 12, you may see the warning `The iOS Simulator deployment target is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.0.99`. This will not prevent your app from compiling. This is a [CocoaPods issue](https://github.com/CocoaPods/CocoaPods/issues/7314) with a known workaround.
+*Note:* If you are using version 8.x.x of the Braintree Drop-in iOS SDK in Xcode 12, you may see the warning `The iOS Simulator deployment target is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.0`. This will not prevent your app from compiling. This is a [CocoaPods issue](https://github.com/CocoaPods/CocoaPods/issues/7314) with a known workaround.
```
Comment (jaxdesmarais) [inline README.md:48]: I don't _think_ we need to change this since it's related to iOS version and not Xcode

File: README.md [hunk 4]
```diff
@@ -235,7 +235,7 @@ Versions 6 and below are unsupported.
 
 ## Demo
 
-A demo app is included in this project. Running the demo app requires Xcode 13+. Open BraintreeDropIn.xcworkspace and run the Demo scheme to view it.
+A demo app is included in this project. Running the demo app requires Xcode 15.1+. Open BraintreeDropIn.xcworkspace and run the Demo scheme to view it.
```
Comment (jaxdesmarais) [inline README.md:238]:
```suggestion
A demo app is included in this project. Running the demo app requires Xcode 15.0+. Open BraintreeDropIn.xcworkspace and run the Demo scheme to view it.
```

File: README.md (hunk 3 above)
Comment (stechiu) [reply README.md]: Ah I missed this one, thanks

**Review by jaxdesmarais — APPROVED** [review]
The SPM failure is due to this work being done off of a fork and the UI tests failure has been consistent across the last few PRs.

**Comment by scannillo** [comment]
Should we also bump the Package.swift ([here](https://github.com/braintree/braintree-ios-drop-in/blob/621ad6debba7b592128049df941e94489e2560e3/Package.swift#L1)) & Podpsec ([here](https://github.com/braintree/braintree-ios-drop-in/blob/621ad6debba7b592128049df941e94489e2560e3/BraintreeDropIn.podspec#L22)) to the newer Swift version too?

Otherwise we can't leverage new language features

Cc: @jaxdesmarais

**Comment by jaxdesmarais** [comment]
> Should we also bump the Package.swift ([here](https://github.com/braintree/braintree-ios-drop-in/blob/621ad6debba7b592128049df941e94489e2560ee3/Package.swift#L1)) & Podpsec ([here](https://github.com/braintree/braintree-ios-drop-in/blob/621ad6debba7b592128049df941e94489e2560e3/BraintreeDropIn.podspec#L22)) to the newer Swift version too?
> 
> Otherwise we can't leverage new language features when building for those packages

Well, we aren't using Swift at all here. 🙈 But yeah I don't see why we couldn't bump those and eventually we can take advantage of those features! @stechiu do you want to bump those in this PR?

**Comment by stechiu** [comment]
> > Should we also bump the Package.swift ([here](https://github.com/braintree/braintree-ios-drop-in/blob/621ad6debba7b592128049df941e94489e2560ee3/Package.swift#L1)) & Podpsec ([here](https://github.com/braintree/braintree-ios-drop-in/blob/621ad6debba7b592128049df941e94489e2560e3/BraintreeDropIn.podspec#L22)) to the newer Swift version too?
> > Otherwise we can't leverage new language features when building for those packages
> 
> Well, we aren't using Swift at all here. 🙈 But yeah I don't see why we couldn't bump those and eventually we can take advantage of those features! @stechiu do you want to bump those in this PR?

Awesome, made the changes so I think we're good now

**Comment by stechiu** [comment]
> We will also want to bump this to Swift 5.9:
> 
> https://github.com/braintree/braintree-ios-drop-in/blob/07807c932c82b36ebc9d5e158bdff17487ef77cc/Package.swift#L1

🤦‍♀️ I did make the change and could've sworn that I merged it in yesterday...

**Review by KunJeongPark — APPROVED** [review]

**Pull request merged**